torchgen: support exception boundary for ExecuTorch functions

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #144341

Needed for ExecuTorch diff D67904052.

Differential Revision: [D67906411](https://our.internmc.facebook.com/intern/diff/D67906411/)

swolchok committed:
torchgen: support exception boundary for ExecuTorch functions

Needed for ExecuTorch diff D67904052.

Differential Revision: [D67906411](https://our.internmc.facebook.com/intern/diff/D67906411/)

[ghstack-poisoned]

swolchok committed:
Update on "torchgen: support exception boundary for ExecuTorch functions"


Needed for ExecuTorch diff D67904052.

Differential Revision: [D67906411](https://our.internmc.facebook.com/intern/diff/D67906411/)

[ghstack-poisoned]

swolchok committed:
Update on "torchgen: support exception boundary for ExecuTorch functions"


Needed for ExecuTorch diff D67904052.

Differential Revision: [D67906411](https://our.internmc.facebook.com/intern/diff/D67906411/)

[ghstack-poisoned]

swolchok committed:
Update on "torchgen: support exception boundary for ExecuTorch functions"


Needed for ExecuTorch diff D67904052.

Differential Revision: [D67906411](https://our.internmc.facebook.com/intern/diff/D67906411/)

[ghstack-poisoned]

Jack-Khuu commented:
This passes the smell test, but it's been a while since I've looked at this code

cc: @larryliu0820 who is will be more familiar

swolchok committed:
Update on "torchgen: support exception boundary for ExecuTorch functions"


Needed for ExecuTorch diff D67904052.

Differential Revision: [D67906411](https://our.internmc.facebook.com/intern/diff/D67906411/)

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
